### ISSUE TITLE
iron-data-table dataSource change

### ISSUE THREAD

**Issue by BartoszGalek** [Saulis/iron-data-table#160]
iron-data-table component has a dataSource property which is a function
![image](https://cloud.githubusercontent.com/assets/4212103/19520993/6781809c-9612-11e6-9c6a-3b7e373c9da5.png)

it also contains _pagesLoading
![image](https://cloud.githubusercontent.com/assets/4212103/19521009/74a48c88-9612-11e6-8a80-edf88d4a1fb5.png)

when the new function is being assigned to dataSource and the page is currently being loaded (slow connection) then observers are being triggered
![image](https://cloud.githubusercontent.com/assets/4212103/19521225/2c2ce5da-9613-11e6-9beb-a78846d947d7.png)

_resetData clears selection, cache but does not clear loading pages
![image](https://cloud.githubusercontent.com/assets/4212103/19521295/60eda278-9613-11e6-80bc-993b1de16af1.png)

as a consequence the _loadPage is calling the previously set (or in my case unset) dataset function and the page is never being loaded
![image](https://cloud.githubusercontent.com/assets/4212103/19521487/23d3cbe6-9614-11e6-97f8-f1f0d4cc5a1b.png)

to put it in more detail:

The problem happens after the initial dataSource was invoked by the table and the page[0] started loading, but before the dataSource returned data to the callback (slow connection). Then the new dataSource is reassigned but the data are (in the above image check "else if" statement) not requested from the new dataSource, since the page is still registered in _pagesLoading.

We believe that the _pagesLoading should be reset when the table gets a new dataSource, along the lines of resetData method.

**Comment by Saulis** [comment]
Yeah I see... there's probably a related problem concerning the `_cachedPages` property which will be appended after the previous data request returns.
